commerce_coordinator/apps/titan/tasks.py

```diff
@@ -102,8 +101,22 @@ def payment_processed_save_task(
             payment_state_paid_cache_key = get_paid_payment_state_cache_key(payment_number)
             TieredCache.set_all_tiers(payment_state_paid_cache_key, payment, settings.DEFAULT_TIMEOUT)
         elif payment_state == PaymentState.FAILED.value:
+            if 'metadata' in provider_response_body:
+                if 'payment_number' in provider_response_body['metadata']:
+                    provider_response_body['metadata']['payment_number'] = ''
```

Comment:
What are we doing here?

Reply:
Removing the old payment number from the stripe response metadata so its not saved in the new payment, I guess it isnt really necessary since well update that information when the user clicks Pay again and isnt used in between those events